### PR TITLE
iox-#961 Use Multi-Arch compliant install destinations

### LINKED ISSUE #961: Comply with Multi-Arch specification for installation on Debian systems
## Brief feature description

The GNUInstallDirs CMake module automatically detects a number of edge cases for FHS-compliant installations. In particular, this makes Iceoryx comply with the Multi-Arch specification on Debian and Debian derivatives by default.

## Detailed information

The proposed change will modify the install location on Debian systems if `CMAKE_INSTALL_PREFIX STREQUAL "/usr"`; in particular, it will add the architecture-triplet to `/usr/lib`, e.g. `/usr/lib/x86_64-linux-gnu`. This is fully supported by CMake and the underlying operating systems, so this change requires no code modifications by Iceoryx users.

### PR DESCRIPTION
**Pull request by roehling**
## Pre-Review Checklist for the PR Author

1. [X] Code follows the coding style of [CONTRIBUTING.md][contributing]
1. [X] Tests follow the [best practice for testing][testing]
1. [x] Changelog updated [in the unreleased section][changelog] including API breaking changes
1. [x] Branch follows the naming format (`iox-#123-this-is-a-branch`)
1. [X] Commits messages are according to this [guideline][commit-guidelines]
    - [X] Commit messages have the issue ID (`iox-#123 commit text`)
    - [X] Commit messages are signed (`git commit -s`)
    - [X] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [X] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [X] Relevant issues are linked
1. [X] Add sensible notes for the reviewer
1. [X] All checks have passed (except `task-list-completed`)
1. [X] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[contributing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#coding-style
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/advanced/best-practice-for-testing.md
[changelog]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/CHANGELOG.md

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Code according to our coding style and naming conventions
- [x] Unit tests have been written for new behavior
- [x] Public API changes are documented via doxygen
- [x] Copyright owner are updated in the changed files
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [x] All open points are addressed and tracked via issues

## References

- Closes #961

### CODE REVIEW

**Comment by elfenpiff** [comment]
@roehling Thanks for your contribution!

We are a project for safety critical systems and therefore we have to follow the traceability principle strictly. Could you please follow the contribution guidelines here: https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md

The essence is:
1. Before creating a PR create an issue first which describes the task you would like to perform (one sentence can sometimes be enough)
2. Create a branch which has the prefix `iox-#???` where the ??? are the number of the issue you created.
3. Every commit requires the prefix `iox-#???`, again with the issue number
4. Please use our Pull Request template
5. The PR name has to have again the `iox-#???` prefix.

I know it is a little bit annoying but this is required to ensure the highest code quality and a later safety certification.

**Review by elfenpiff — CHANGES_REQUESTED** [review]
The contents of the PR are fine but sadly I have to insist that you comply to the contribution guidelines.

**Comment by roehling** [comment]
PR has been updated.

File: iceoryx_hoofs/cmake/IceoryxPackageHelper.cmake [hunk 1]
```diff
@@ -18,6 +18,7 @@
 # setup_package_name_and_create_files : this macro which is called from other modules which use iceoryx_hoofs
 # sets the variables for package version file,config file used for configuration
 # this also creates the config files
+include(GNUInstallDirs)
```
Comment (elfenpiff) [inline iceoryx_hoofs/cmake/IceoryxPackageHelper.cmake:21]: Does this also work on MacOS and Windows? I am asking since we are supporting those platforms.

Comment (roehling) [reply iceoryx_hoofs/cmake/IceoryxPackageHelper.cmake:21]: Yes, it does. By default, the module uses the GNU conventions, which happen to coincide with the previous hard-coded values. The special cases such as multi-arch lib locations are only triggered for system-wide installation to prefix `/usr` or `/`, and only on certain Linux distributions which support these.

**Review by elfenpiff — DISMISSED** [review]

**Comment by dkroenke** [comment]
@roehling 
Would add an entry in the unreleased section for https://github.com/eclipse-iceoryx/iceoryx/issues/960 https://github.com/eclipse-iceoryx/iceoryx/issues/961 and https://github.com/eclipse-iceoryx/iceoryx/issues/962 ? 

I guess this can go into the bugfix section and don't forget to add your name to it :grin:

**Comment by roehling** [comment]
I'll add the changelog entry for #960 to this PR, as the former has been merged already.

**Review by dkroenke — APPROVED** [review]

**Review by elfenpiff — APPROVED** [review]

**Pull request merged**